flaxandteal-arches/crimson
# Feat: Import Single CSV with Processing Vue ETL Module

## Description
This PR creates an ETL module for the data bulk manager.
The module is built using Vue and has been modified to work with the bulk data manager which is running in knockout.
This current module is a clone of the single import csv and serves as a base to build additional features on top of.

## Features
The Vue is split into 3 components:
- ImportSingleCSVwithProcessing.vue - this controls the module and its state and child components
- TaskDetailsComponent.vue - this handles the 'task details' tab of the data manager and allows for initial preview of the csv
- TaskStatusComponent.vue - this handles the information passed to the task status tab of the data manager

There is an additional shareState.js - this allows for state to be shared between the ko data manager and the vue module

 Python files:
- import_single_csv_with_processing.py - this currently is a clone of import_single_csv.py and has working functionality to allow data to be imported (this is to be updated)

## Additional
The task status 'Validation Errors' is still controlled via a script in the data manager.
This could be converted to Vue and will be in future updates if time allows for it.

## Test
- You need to import the module:
```
make manage CMD="etl_module register -s crimson/etl_modules/import_single_csv_with_processing.py"
```

- activate the data manager, follow the instructions here:
https://arches.readthedocs.io/en/stable/administering/bulk-data-manager/#bulk-data-manager

- Navigate to the data manager plugin
- You should see the 'Import Single CSV with Processing' module
- Click this and it should bring up a browse button
- The browse button should allow you to import a csv
[data-extracts-objects.csv](https://github.com/user-attachments/files/17936738/data-extracts-objects.csv)
- Once imported you should see a drop down to allow you to select a model
- This should show a table with a preview of your csv data
- Select several nodes that match the headings and hit submit
- This should take you to the task status page where you should see your task and whether the data has validated
- You should see information about the upload and any errors that may have occurred with the data
- If the upload was successful, head to the arches search and you should see your data
